pyiron_nodes/atomistic/engine/vasp.py

```diff
@@ -0,0 +1,263 @@
+    _ = write_POTCAR(workdir=workdir, vasp_input=vasp_input)
+    if vasp_input.kpoints is not None:
+        _ = write_KPOINTS(workdir=workdir, kpoints=vasp_input.kpoints)
+
+    return workdir
+
+
+class Storage:
+    @staticmethod
+    def _convert_to_dict(instance) -> Dict:
+        attributes = vars(instance)
+        result_dict = {
+            key: value for key, value in attributes.items() if "_" not in key[0]
+        }
+        return result_dict
+
+
+class ShellOutput(Storage):
+    stdout: str
+    stderr: str
+    return_code: int
+
+
+@Workflow.wrap.as_function_node("output")
+def run_job(
+    command: str,
+    workdir: str = os.getcwd(),
+    environment: Optional[Dict[str, str]] = None,
+    arguments: Optional[List[str]] = None,
+) -> ShellOutput:
+    if environment is None:
+        environment = {}
+    if arguments is None:
+        arguments = []
+    print(f"run_job is in {os.getcwd()}")
+    environ = dict(os.environ)
+    environ.update({k: str(v) for k, v in environment.items()})
+    proc = subprocess.run(
+        [command, *map(str, arguments)],
+        capture_output=True,
+        cwd=workdir,
+        encoding="utf8",
+        env=environ,
+        shell=True,
+    )
+    print(f"Running job in directory: {workdir}")
+    output = ShellOutput()
+    output.stdout = proc.stdout
+    output.stderr = proc.stderr
+    print(proc.stdout, proc.stderr)
+    output.return_code = proc.returncode
+    return output
+
+
+@Workflow.wrap.as_function_node("output_dict")
+def parse_VaspOutput(workdir: str) -> Dict:
+    print(f"workdir of parse: {workdir}")
+    return pvo(workdir)
+
+
+@Workflow.wrap.as_function_node("convergence")
+def check_convergence(
+    workdir: str,
+    filename_vasprun: str = "vasprun.xml",
+    filename_vasplog: str = "vasp.log",
+    backup_vasplog: str = "error.out",
+) -> bool:
+    converged = False
+    line_converged = (
+        "reached required accuracy - stopping structural energy minimisation"
+    )
+
+    try:
+        vr = Vasprun(filename=os.path.join(workdir, filename_vasprun))
+        converged = vr.converged
+    except:
+        try:
+            converged = is_line_in_file(
+                filepath=os.path.join(workdir, filename_vasplog),
+                line=line_converged,
+                exact_match=False,
+            )
+        except:
+            try:
+                converged = is_line_in_file(
+                    filepath=os.path.join(workdir, backup_vasplog),
+                    line=line_converged,
+                    exact_match=False,
+                )
+            except:
+                pass
+
+    return converged
+
+
+@Workflow.wrap.as_macro_node("vasp_output", "convergence_status")
+def vasp_job(
+    self,
+    workdir: str,
+    vasp_input,
+    command: str = "module load vasp; module load intel/19.1.0 impi/2019.6; unset I_MPI_HYDRA_BOOTSTRAP; unset I_MPI_PMI_LIBRARY; mpiexec -n 1 vasp_std",
```

Comment:
I think it's an overreach to try and fit it in this PR, but seeing this I'm immediately struck whether module loads should be a built-in feature of something like `run_job`, in the same way that [my scipy stuff exposes mpi choices](https://github.com/liamhuber/scipy2024/blob/6c047f5b5e3a7f7f803b3c6cdb7768891582b7c5/ase_nodes/calculators.py#L140), e.g. an input `modules: list[str] | None` and a wrapper function that prepends the actual command with all the `"module load {mod}"` pieces.